osutil/disks/mockdisk.go

```diff
@@ -0,0 +1,144 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+var (
+	isSnapdTest = len(os.Args) > 0 && strings.HasSuffix(os.Args[0], ".test")
+)
+
+func panicIfNotTest() {
+	// only to be used in tests!!!!
+	if !isSnapdTest {
+		panic("mocking functions only to be used in tests!")
+	}
+}
+
+// MockDiskMapping is an implementation of Disk for mocking purposes, it is
+// exported so that other packages can easily mock a specific disk layout
+// without needing to mock the mount setup, sysfs, or udev commands just to test
+// high level logic. We use an embedded type of an unexported type so as not to
```

Comment:
I don't understand the reference to an embedded type of an unexported type. Does it refer to some previous impl?

Reply:
sorry yes it does refer to a previous impl, let me delete that

Reply:
You marked this resolved but the comment is still there??

Reply:
hmm that's odd I definitely deleted it, I must have messed something up when I rebased this on top of the other xerrors PR, sorry I will re-fix that again

Reply:
ok now it's gone for good :+1: :+1: :+1: